cmd/otel-allocator/allocation/consistent_hashing_test.go

```diff
@@ -17,12 +17,34 @@ package allocation
 import (
 	"testing"
 
+	"github.com/prometheus/prometheus/model/relabel"
 	"github.com/stretchr/testify/assert"
+
+	"github.com/open-telemetry/opentelemetry-operator/cmd/otel-allocator/prehook"
+	"github.com/open-telemetry/opentelemetry-operator/cmd/otel-allocator/target"
 )
 
+var _ prehook.Hook = &mockHook{}
+
+type mockHook struct {
+}
+
+func (hook mockHook) Apply(targets map[string]*target.Item) map[string]*target.Item {
+	return targets
+}
+
+func (hook mockHook) SetConfig(map[string][]*relabel.Config) {
+}
+
+func (hook mockHook) GetConfig() map[string][]*relabel.Config {
+	dummyRet := map[string][]*relabel.Config{}
+	return dummyRet
+}
```

Comment:
These don't seem to be at all relevant to the use of the `prehook.Hook` type here.  Can that interface have just the `Apply()` method and leave `[SG]etConfig()` to the concrete implementation of the relabeling hook?

Reply:
Hmm I'm still unsure the best way to implement this suggestion. I agree that `[SG}etConfig()` methods are specific to relabeling filter but the`relabelCfg` gets set in discovery.  So I'm actually unsure if I'm able to get access to the concrete struct's method, when I only pass in the allocator interface.